### PR TITLE
fix: config form values reset

### PR DESCRIPTION
**Pull request by marcgarreau**
#### What does it do?
- fixes a bug where deleting config form values reset them to defaults.
- cleans up some incorrect logic for updating settings (e.g. in Nodes/index.js, we set `selectedClient.selectedConfig = newConfig`, but never did anything with `selectedConfig`)
- abstracts out `getPluginSettingsConfig` to utils and tests it.
- relocates the `generateFlags` util function to the same utils file.
#### Does it close any issues?
Closes https://github.com/ethereum/grid/issues/231
Closes https://github.com/ethereum/grid/issues/210 (i think this was already addressed)

### CODE REVIEW

**Comment by evertonfraga** [comment]
LGTM, will test tomorrow morning

File: src/components/Nodes/ClientConfig/index.js [hunk 1]
```diff
@@ -131,23 +131,29 @@ class ClientConfig extends Component {
...
-              settings={getPluginSettingsConfig(client)}
-              handleClientConfigChanged={this.handleClientConfigChanged}
-              isClientRunning={isRunning}
-              clientConfigChanged={clientConfigChanged}
-            />
-          </ErrorBoundary>
-        </TabContainer>
+
+        {/* NOTE: MUI requires generating the ConfigForm from state each render */}
+        {activeTab === 1 && (
```
Comment (evertonfraga) [inline src/components/Nodes/ClientConfig/index.js]: nice hack, huh?

Comment (marcgarreau) [reply src/components/Nodes/ClientConfig/index.js]: Actually, this is good practice, conditionally rendering the component. The other two tabs may be considered 'hacks' for rendering upfront and showing/hiding the contents via css for performance.

**Comment by marcgarreau** [comment]
@evertonfraga 500ms debounce implemented:

**Pull request merged**